projects/packages/scheduled-updates/src/class-scheduled-updates.php

```diff
@@ -61,23 +64,119 @@ public static function load_rest_api_endpoints() {
 	 * @param string ...$plugins List of plugins to update.
 	 */
 	public static function run_scheduled_update( ...$plugins ) {
+		$schedule_id       = self::generate_schedule_id( $plugins );
 		$available_updates = get_site_transient( 'update_plugins' );
 		$plugins_to_update = $available_updates->response ?? array();
 		$plugins_to_update = array_intersect_key( $plugins_to_update, array_flip( $plugins ) );
 
 		if ( empty( $plugins_to_update ) ) {
+			// No updates available. Update the status to 'success' and return.
+			self::set_scheduled_update_status( $schedule_id, time(), 'success' );
+
 			return;
 		}
 
 		( new Connection\Client() )->wpcom_json_api_request_as_blog(
-			sprintf( '/sites/%d/hosting/scheduled-update', \Jetpack_Options::get_option( 'id' ) ),
+			sprintf( '/sites/%d/hosting/scheduled-update/%s', \Jetpack_Options::get_option( 'id' ), $schedule_id ),
 			'2',
 			array( 'method' => 'POST' ),
 			array( 'plugins' => $plugins_to_update ),
 			'wpcom'
 		);
 	}
 
+	/**
+	 * Updates last status of a scheduled update.
+	 *
+	 * @param string $schedule_id Request ID.
+	 * @param int    $timestamp   Timestamp of the last run.
+	 * @param string $status      Status of the last run.
+	 * @return bool|mixed Updated schedule or false if not found.
+	 */
+	public static function set_scheduled_update_status( $schedule_id, $timestamp, $status ) {
+		$events = wp_get_scheduled_events( 'jetpack_scheduled_update' );
+
+		if ( empty( $events[ $schedule_id ] ) ) {
+			// Scheduled update not found.
+			return false;
+		}
+
+		$statuses = get_option( 'jetpack_scheduled_update_statuses', array() );
+		$option   = array();
+
+		// Reset the last statuses for the schedule.
+		foreach ( array_keys( $events ) as $status_id ) {
+			if ( ! empty( $statuses[ $status_id ] ) ) {
+				$option[ $status_id ] = $statuses[ $status_id ];
+			} else {
+				$option[ $status_id ] = null;
+			}
+		}
+
+		// Update the last status for the schedule.
+		$option[ $schedule_id ] = array(
+			'last_run_timestamp' => $timestamp,
+			'last_run_status'    => $status,
+		);
+
+		update_option( 'jetpack_scheduled_update_statuses', $option );
+
+		return $option[ $schedule_id ];
+	}
+
+	/**
+	 * Get the last statuses of a scheduled updates.
+	 *
+	 * @return array|null Last statuses of the scheduled updates.
+	 */
+	public static function get_scheduled_update_statuses() {
+		return get_option( 'jetpack_scheduled_update_statuses', array() );
```

Comment:
Let's avoid a wrapper when it doesn't do anything. I think it's fine to call the option directly and it'll be easier to debug later.

Reply:
Make sense, removed.